standard/expressions.md

```diff
@@ -1821,6 +1822,49 @@ A  *null_conditional_member_access* expression `E` is of the form `P?.A`. The me
 
 A *null_conditional_projection_initializer* is a restriction of *null_conditional_member_access* and has the same semantics. It only occurs as a projection initializer in an anonymous object creation expression ([§12.8.16.7](expressions.md#128167-anonymous-object-creation-expressions)).
 
+### §Null-Forgiving-Expressions Null-forgiving expressions
+
+This operator sets the null state (§Nullabilities-And-Null-States) of the operand to “not null”.
+
+```ANTLR
+null_forgiving_expression
+    : primary_no_array_creation_expression suppression
+    ;
+
+suppression
+    : '!'
+    ;
+```
+
+The *primary_expression* in *null_forgiving_expression* shall not be a *null_forgiving_expression*.
+
+This operator has no runtime effect; it evaluates to the result of its operand, and that result retains that operand’s classification.
+
+The null-forgiving operator is used to declare that an expression having a reference type isn't null.
```

Comment:
Can it be used on expressions whose type is a generic type parameter not known to be a reference type? I think so, and that would mean that the "having a reference type" isn't always true.